ibm_satellite_location provider crashed when passed an invalid managed_from value

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

```
Terraform v1.13.0
on linux_amd64
+ provider registry.terraform.io/alekc/kubectl v2.1.3
+ provider registry.terraform.io/hashicorp/external v2.3.5
+ provider registry.terraform.io/hashicorp/helm v2.17.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.34.0
+ provider registry.terraform.io/hashicorp/null v3.2.4
+ provider registry.terraform.io/hashicorp/random v3.7.2
+ provider registry.terraform.io/ibm-cloud/ibm v1.81.1
+ provider registry.terraform.io/sysdiglabs/sysdig v1.28.5
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_satellite_location

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "ibm_satellite_location" "satellite_location" {
  location          = "mysatellite-location"
  zones             = ["zone-1", "zone-2", "zone-3"]
  managed_from      = "es"
  resource_group_id = "default"
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

https://gist.github.com/pcbye/511a15c1a4a71d7730bdbe751d051935

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```
panic: runtime error: slice bounds out of range [:3] with length 2
```

### Expected Behavior

<!--- What should have happened? --->

Provider plan or apply should have failed with an appropriate error.

### Actual Behavior

<!--- What actually happened? --->

Provider crashed with the above panic, a stack trace, and a message indicating that `This is always indicative of a bug within the plugin.`

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply` (to create the resource - in my case, it created successfully with this invalid string)
2. `terraform plan` (fails here, re-reading the state of the resource)

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/service/satellite/resource_ibm_satellite_location.go
+++ b/ibm/service/satellite/resource_ibm_satellite_location.go
@@ -87,7 +87,9 @@ func ResourceIBMSatelliteLocation() *schema.Resource {
 				DiffSuppressFunc: func(k, o, n string, d *schema.ResourceData) bool {
 					if o == "" {
 						return false
-					} else if o[0:3] == n[0:3] {
+					} else if len(o) >= 3 && len(n) >= 3 && o[0:3] == n[0:3] {
+						return true
+					} else if len(o) >= 2 && len(n) >= 2 && o[0:2] == n[0:2] {
 						return true
 					}
 					return o == n

